src/classes/groundplaces.ts

```diff
@@ -201,31 +194,119 @@ export class GroundPlacesController {
+   * @returns {void}
+   */
+  public addStopGroupToStopCluster(stopGroupGpuidToAdd: StopGroupGpuid, intoStopClusterGpuid: StopClusterGpuid): void {
+    // Check if the StopGroup specified exist in the Ground places
+    this.storageService.getStopGroupByGpuid(stopGroupGpuidToAdd);
+
+    const stopClusterParent: StopCluster = this.storageService.getStopClusterByGpuid(intoStopClusterGpuid);
+
+    // Check if the StopGroup does not already exist in the new StopCluster parent
+    const isStopGroupAlreadyExist = Boolean(
+      stopClusterParent.childs.find((stopGroupGpuid: StopGroupGpuid) => stopGroupGpuid === stopGroupGpuidToAdd),
+    );
+
+    if (isStopGroupAlreadyExist) {
+      throw new Error(
+        `The StopGroup with the Gpuid "${stopGroupGpuidToAdd}" cannot be added to the StopCluster with the Gpuid "${intoStopClusterGpuid}" because it already exist in it.`,
+      );
+    }
+
+    stopClusterParent.childs.push(stopGroupGpuidToAdd);
+
+    // Update the place with the new StopCluster parent
+    this.storageService.replacePlace(stopClusterParent);
+  }
+
+  /**
+   * @description Remove a stopGroup from a stopCluster.
+   *
+   * Warning: The StopGroup cannot be without StopCluster parent after this operation.
+   * @param {StopGroupGpuid} stopGroupGpuidToRemove - Ground place unique identifier of the SopGroup to remove.
+   * @param {StopClusterGpuid} stopClusterGpuidParent - Ground place unique identifier of the StopCluster parent.
+   * @returns {void}
+   */
+  public removeStopGroupFromStopCluster(
+    stopGroupGpuidToRemove: StopGroupGpuid,
+    stopClusterGpuidParent: StopClusterGpuid,
+  ): void {
+    const stopClusterParent: StopCluster = this.storageService.getStopClusterByGpuid(stopClusterGpuidParent);
+    const groundPlaces: GroundPlace[] = this.storageService.getGroundPlaces();
+    const copyGroundPlaces: GroundPlace[] = this.storageService.cloneGroundPlaces();
+
+    const stopGroupIndex: number = stopClusterParent.childs.findIndex(
+      (stopGroupGpuid: StopGroupGpuid) => stopGroupGpuid === stopGroupGpuidToRemove,
+    );
+
+    if (stopGroupIndex === -1) {
+      throw new Error(
+        `The StopGroup with the Gpuid "${stopGroupGpuidToRemove}" cannot be removed from the StopCluster with the Gpuid "${stopClusterGpuidParent}" because it does not belong to it.`,
+      );
+    }
+
+    // Remove the reference of the StopGroup Gpuid inside the StopCluster parent
+    stopClusterParent.childs.splice(stopGroupIndex, 1);
+
+    // Update the place with the new StopCluster parent
+    this.storageService.replacePlace(stopClusterParent);
+
+    // Check if the StopGroup is not lonely after the operation
+    const stopGroupExist = groundPlaces.find(
+      (groundPlace: GroundPlace): boolean =>
+        groundPlace.type === GroundPlaceType.CLUSTER &&
+        Boolean(groundPlace.childs.find((stopGroupGpuid) => stopGroupGpuid === stopGroupGpuidToRemove)),
```

Comment:
why `Boolean()` ? maybe check the `.some()` method which returns a boolean.

Reply:
Yes thanks! Missed this one little guy :)